components/raftstore/src/store/worker/split_controller.rs

```diff
@@ -826,12 +826,18 @@ impl AutoSplitController {
             if let Some(hottest_key_range) = hottest_key_range {
                 recorder.update_hottest_key_range(hottest_key_range);
             }
+            let key_ranges = vec![];
+            if let Some(hottest_key_range) = hottest_key_range {
+                recorder.update_hottest_key_range(hottest_key_range);
+                key_ranges = vec![hottest_key_range];
```

Comment:
One risk of doing this is that, due to one less sampling, the final number of regions to be split may increase. In this case, when a hotspot appears, increasing the number of regions may bring additional side effects, such as increasing the number of tasks or gRPC poll times, thereby affecting performance. We should conduct some tests on this change before merging it.

Reply:
ok, i raised this github issue https://github.com/tikv/tikv/pull/17010 only for logging. Let me know how it looks.

Reply:
Why are you using the hottest key range inside top_cpu_usage(). Shouldn't we do sampling over there.